packages/common/src/TableId.ts

```diff
@@ -0,0 +1,37 @@
+import { Hex, stringToHex, hexToString, sliceHex, concatHex } from "viem";
+
+export class TableId {
```

Comment:
should we rename this to `ResourceSelector`? Or have a base class `ResourceSelector` this inherits from? (See https://github.com/latticexyz/mud/pull/999#discussion_r1222208895)

Reply:
ohh I like that better, will do

Reply:
hmm, gonna save this for later! in searching the codebase for "tableid", we have a lot of instances and so might wanna rethink the boundaries of this